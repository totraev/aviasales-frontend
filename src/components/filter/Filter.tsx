import React, { SFC } from 'react';
import styles from './Filter.module.css';

import CurrencyToggle from '../../containers/currencyToggle/CurrencyToggle';
import StopsFilter from '../../containers/stopsFilter/StopsFilter';

const Filter: SFC = () => (
  <div className={styles.filter}>
    <h3 className={styles.title}>ВАЛЮТА</h3>
    <CurrencyToggle
      currency="usd"
      onChange={null}
    />

    <h3 className={styles.title}>КОЛИЧЕСТВО ПЕРЕСАДОК</h3>

    <StopsFilter stopsList={[true, true, true, true]}/>
  </div>
);

export default Filter;
