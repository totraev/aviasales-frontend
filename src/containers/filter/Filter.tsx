import React, { SFC } from 'react';
import styles from './Filter.module.css';

import CurrencyToggle from '../../components/currencyToggle/CurrencyToggle';
import StopsFilter from '../../components/stopsFilter/StopsFilter';

const Filter: SFC = () => (
  <div className={styles.filter}>
    <h3 className={styles.title}>ВАЛЮТА</h3>
    <CurrencyToggle/>

    <h3 className={styles.title}>КОЛИЧЕСТВО ПЕРЕСАДОК</h3>

    <StopsFilter />
  </div>
);

export default Filter;
