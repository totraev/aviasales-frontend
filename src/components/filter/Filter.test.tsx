import React from 'react';
import renderer from 'react-test-renderer';

import Filter from './Filter';

describe('Filter components', () => {
  it('Should renders correctly', () => {
    const tree = renderer.create(<Filter />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
