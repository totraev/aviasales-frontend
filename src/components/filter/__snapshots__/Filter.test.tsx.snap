// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter components Should renders correctly 1`] = `
<div
  className="filter"
>
  <h3
    className="title"
  >
    ВАЛЮТА
  </h3>
  <div
    className="toggle"
  >
    <button
      className="btnDefault"
      onClick={[Function]}
    >
      RUB
    </button>
    <button
      className="btnActive"
      onClick={[Function]}
    >
      USD
    </button>
    <button
      className="btnDefault"
      onClick={[Function]}
    >
      EUR
    </button>
  </div>
  <h3
    className="title"
  >
    КОЛИЧЕСТВО ПЕРЕСАДОК
  </h3>
  <div
    className="stops"
  >
    <div
      className="wrap"
    >
      <label
        className="label"
        htmlFor="stops_0"
      >
        <input
          checked={true}
          className="checkbox"
          id="stops_0"
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="faceChecked"
        />
        БЕЗ ПЕРЕСАДОК
      </label>
      <div
        className="append"
      >
        <button
          className="uncheckOther"
          onClick={[Function]}
        >
          ТОЛЬКО
        </button>
      </div>
    </div>
    <div
      className="wrap"
    >
      <label
        className="label"
        htmlFor="stops_1"
      >
        <input
          checked={true}
          className="checkbox"
          id="stops_1"
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="faceChecked"
        />
        1 ПЕРЕСАДКА
      </label>
      <div
        className="append"
      >
        <button
          className="uncheckOther"
          onClick={[Function]}
        >
          ТОЛЬКО
        </button>
      </div>
    </div>
    <div
      className="wrap"
    >
      <label
        className="label"
        htmlFor="stops_2"
      >
        <input
          checked={true}
          className="checkbox"
          id="stops_2"
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="faceChecked"
        />
        2 ПЕРЕСАДКИ
      </label>
      <div
        className="append"
      >
        <button
          className="uncheckOther"
          onClick={[Function]}
        >
          ТОЛЬКО
        </button>
      </div>
    </div>
    <div
      className="wrap"
    >
      <label
        className="label"
        htmlFor="stops_3"
      >
        <input
          checked={true}
          className="checkbox"
          id="stops_3"
          onClick={[Function]}
          type="checkbox"
        />
        <span
          className="faceChecked"
        />
        3 ПЕРЕСАДКИ
      </label>
      <div
        className="append"
      >
        <button
          className="uncheckOther"
          onClick={[Function]}
        >
          ТОЛЬКО
        </button>
      </div>
    </div>
  </div>
</div>
`;
